HW27/Vector2d.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <cmath>
+
+struct Vector2d
+{
+    int x = 0;
+    int y = 0;
+
+    // i. ����������� ��� �������� ������� � ���� �����: tail, head
+    Vector2d(float x0, float y0, float x1, float y1)
+    {
+        x = static_cast<int>(x1 - x0);
+        y = static_cast<int>(y1 - y0);
+    }
+
+    // ii. ��������� �������
+    float dotProduct(const Vector2d& other) const
+    {
+        return static_cast<float>(x * other.x + y * other.y);
+    }
+
+    // iii. ������� (���������� ������)
+    Vector2d negate() const
+    {
+        return Vector2d(-x, -y);
+    }
+
+    // iv. Enum class ��� ���������� �������� ��������� �������
+    enum class VectorRelativeState
+    {
+        Identical,
+        CoDirected,
+        OppositeDirected,
+        AcuteAngle,
+        ObtuseAngle,
+        RightAngle
+    };
+
+    // iv. ����� ��� ���������� �������� ��������� �������
+    VectorRelativeState getRelativeState(const Vector2d& other) const
+    {
+        float dot = dotProduct(other);
+        float magnitude1 = std::sqrt(static_cast<float>(x * x + y * y));
+        float magnitude2 = std::sqrt(static_cast<float>(other.x * other.x + other.y * other.y));
+        float cosTheta = dot / (magnitude1 * magnitude2);
```

Comment:
can be calculated without cosine operation. See Lecture slides for reference using only dotProduct value. In general, trig and sqrt functions are heavy relative to mult, so better avoid it if possible